sys/src/9/ufs/ufs_harvey.h

```diff
@@ -120,16 +99,21 @@ typedef enum vtype Vtype;
  * It may be unnecessary - if so, we should use inodes directly.
  */
 typedef struct vnode {
+	vnode		*next;
+	vnode		*prev;
+
+	Ref		ref;		/* Refcount */
+	QLock		v_vnlock;	/* u pointer to vnode lock */
+
 	/*
 	 * Fields which define the identity of the vnode.  These fields are
 	 * owned by the filesystem (XXX: and vgone() ?)
 	 */
-	const char 	*v_tag;			/* u type of underlying data */
 	inode		*v_data;
 	MountPoint	*v_mount;
```

Comment:
general comment here and elsewhere. 

FWIW the plan 9 style tended in later years to CamelCase, not camel_case. I.e. more like go, less like traditional Unix. Also, for example, the v->v_mount struct element thing was gradually discarded in favor of just the name, i.e. v_mount would become just Mount or mount. The v_ type stuff really goes back to the time when any struct pointer could deref any struct type (true! see dmr's early paper in the v6 docs). At some point I'd favor dropping all the v_ etc. stuff in these structs but that's your call of course.